### ISSUE TITLE
zls installation fails on Windows

### ISSUE THREAD

**Issue by DasPez** [williamboman/mason.nvim#947]
There is a typo in the zls install script.

`when(platform.is.win_x64, "i386-windows.tar.zst"),` should be `platform.is.win_x86`

https://github.com/williamboman/mason.nvim/blob/b9e8c4a1130f800e1c3b86be207ff4d5b49218c0/lua/mason-registry/zls/init.lua#L24

**Comment by williamboman** [comment]
Yeah that doesn't look quite right! PR welcome, otherwise I'll get around to it asap

**Comment by DasPez** [comment]
The installation still fails. I think the cause is the tar implementation shipped with Windows. It is based on libarchive and does not support .zst decompression by itself.
```
> tar --version
bsdtar 3.5.2 - libarchive 3.5.2 zlib/1.2.5.f-ipp bz2lib/1.0.6
```
I tried running `> tar -xvf .\x86_64-windows.tar.zst` in powershell and it did not terminate at all.
I had to install zstandard and manually decompress it first with
`> zstd -d .\x86_64-windows.tar.zst` to be able to run
`> tar -xvf .\x86_64-windows.tar`

Is there a graceful way to pull in the zstandard dependency needed for the installation on windows?
